packages/protocol/contracts/L1/libs/LibVerifying.sol

```diff
@@ -88,6 +88,10 @@ library LibVerifying {
                 fc.prover = address(0);
                 fc.provenAt = 0;
             }
+
+            unchecked {
+                ++i;
```

Comment:
I [tested](https://github.com/taikoxyz/taiko-mono/blob/94af89a51da034f34a673a9ea97ee381e0674277/packages/protocol/test2/GasComparison.t.sol#L97), `i++` inside unchecked is cheaper.

| test2/GasComparison.t.sol:FooBar contract |                 |        |        |        |         |
|-------------------------------------------|-----------------|--------|--------|--------|---------|
| Deployment Cost                           | Deployment Size |        |        |        |         |
| 187432                                    | 968             |        |        |        |         |
| Function Name                             | min             | avg    | median | max    | # calls |
| hashString_1                              | 1037            | 1037   | 1037   | 1037   | 1       |
| hashString_2                              | 1015            | 1015   | 1015   | 1015   | 1       |
| hashTwo_1                                 | 438             | 438    | 438    | 438    | 1       |
| hashTwo_2                                 | 511             | 511    | 511    | 511    | 1       |
| increment_1                               | 185215          | 185215 | 185215 | 185215 | 1       |
| increment_2                               | 180353          | 180353 | 180353 | 180353 | 1       |
| increment_3                               | 119347          | 119347 | 119347 | 119347 | 1       |
| increment_4                               | 119390          | 119390 | 119390 | 119390 | 1       |

Reply:
That is actually a really funny one ! 🙂
 It is not related to the pre or post increment - tho it looks like !
 
The reason behind is the function hashes. The compiler always orders the function hashes 
```
 "552629ac": "increment_1(uint256)",
 "514525f6": "increment_2(uint256)",
 "1752e353": "increment_3(uint256)",
  "ebf32ef2": "increment_4(uint256)"
```
 and since the `17..` is always before `eb..`, alseo the `increment_1` and `increment_2` the `eb..` requires more opcodes to be called.
 
Generally if a function call is very gas sensitive, function selector should be near the top. Also if we benchmark a function, we shall not change the name - so best would be to test in-place (with existing unit tests) to check which one is cheaper.

If you swap the 2 names now in your tests, you would see the `increment_3()` would be always the cheaper.

(BTW after 0.8.X there might be something like the compiler does optimize the ++i and i++ the same way, but i need to double check / benchmark with our current unit tests.)

Reply:
Did the benchmarking. 

With post increment:
<img width="482" alt="kép" src="https://user-images.githubusercontent.com/51912515/223693335-69cd72af-2e73-4722-b53d-dc1ef5a44aaf.png">

With pre increment:
<img width="482" alt="kép" src="https://user-images.githubusercontent.com/51912515/223693367-1afc4ab0-719f-426f-8c91-6cabadff65ac.png">

As proveBlock + proposeBlock calls into verifyBlock, we can see some tiny savings there, but that might be because of other factors (chain state, blocknumber, etc.). May best guess is that the compiler optimizes it as much as possible and after 0.8.X as mentioned, there is no difference.

Conclusion, we can use whichever, diff is negligible.